scripts/generate_release_notes.py

```
@@ -0,0 +1,330 @@
+            
+            current_version = resolver.get_current_version()
+            latest_tag = resolver.get_latest_git_tag() or "0.0.0"
+            
+            # Get commits since last release
+            commits = resolver.get_commits_since_version(latest_tag)
+            
+            if commits:
+                # Auto-suggest version based on commits
+                suggested_version, bump_type, impact_scores = resolver.suggest_version_bump(current_version, commits)
+                
+                print(f"📊 Smart Version Analysis:")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)